test/kube_checks.go

```diff
@@ -97,26 +97,27 @@ func WaitForPodState(client *KubeClient, inState func(p *corev1.Pod) (bool, erro
 // from client every interval until number of service endpoints = numOfEndpoints
 func WaitForServiceEndpoints(client *KubeClient, svcName string, svcNamespace string, numOfEndpoints int) error {
 	endpointsService := client.Kube.CoreV1().Endpoints(svcNamespace)
-	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForServiceHasAtLeastOneEndpoint/%s", svcName))
+	span := logging.GetEmitableSpan(context.Background(), "WaitForServiceHasAtLeastOneEndpoint/"+svcName)
 	defer span.End()
 
 	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
-		endpoints, err := endpointsService.List(metav1.ListOptions{})
+		endpoint, err := endpointsService.Get(svcName, metav1.GetOptions{})
 		if err != nil {
-			return true, err
+			return false, err
 		}
 
-		for _, e := range endpoints.Items {
-			if e.Name == svcName && countEndpointsNum(&e) == numOfEndpoints {
-				return true, nil
-			}
+		if countEndpointsNum(endpoint) == numOfEndpoints {
```

Comment:
and the final change, please
`return countEndpointsNum(endpoint) == numOfEndpoints, nil`